reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/LiteralToUniformReductionOpportunity.java

```diff
@@ -25,18 +30,40 @@
 
   private final Expr literalExpr;
   private final PipelineInfo pipelineInfo;
+  private final TranslationUnit translationUnit;
+  private final int index;
 
-  LiteralToUniformReductionOpportunity(Expr literalExpr,
+  LiteralToUniformReductionOpportunity(TranslationUnit translationUnit,
+                                       Expr literalExpr,
+                                       int index,
                                        PipelineInfo pipelineInfo,
                                        VisitationDepth depth) {
     super(depth);
+
+    this.index = index;
     this.literalExpr = literalExpr;
     this.pipelineInfo = pipelineInfo;
+    this.translationUnit = translationUnit;
+  }
+
+  public Expr getLiteral() {
+    return this.literalExpr;
+  }
+
+  public int getIndex() {
+    return this.index;
   }
 
   @Override
   void applyReductionImpl() {
-    // TODO: ...
+
```

Comment:
Here, you should check whether the required array already exists in the shader and pipeline info.  If it doesn't, you should bring the array into existence.

Reply:
The code you have currently in the constructor should go here.

What are currently fields should be local variables.